### PR TITLE
HDRP support

### PR DESCRIPTION
**Pull request by EMBYRDEV**
This isn't complete as I need to fix some alpha issues (however I have 0 experience writing shaders with unity and HDRP) and add support for procedural rendering.

One of the things that seems strange to me is that alpha blending works perfectly when injected "Before Transparency" but it affects eye adaptation and stops the rendering of transparent objects. Probably a fairly simple fix but we really want it to be injected after post processing which is where issues occur.

If one of the devs here would be willing to look into this you'll probably have more success than me but for now this is a perfectly functional port to the HDRP and this PR could help some people who need support ASAP and get the ball rolling on 100% accurate support.

### CODE REVIEW

**Comment by EMBYRDEV** [comment]
Quick demo screenshot.

image

**Comment by nothke** [comment]
Thanks for that! But I'm not exactly sure how to set it up with HDRP. The "render feature" part of the docs confuse me. Does that even exist in HDRP?

**Comment by EMBYRDEV** [comment]
@nothke no, that is a URP feature. You can check out the updated read me with HDRP setup [here](https://github.com/realgamessoftware/dear-imgui-unity/pull/11/commits/f23b704c2b75ab0e8068ddad719ccdd5aa8878e8?short_path=04c6e90#diff-04c6e90faac2675aa89e2176d2eec7d8).

Just right click in your scene, add a custom render pass volume and add a Dear ImGui pass after post processing. Make sure you actually pull down this pull request as this isnt actually integrated into the main repo yet, it's a request for the developers to do so :)

**Comment by nothke** [comment]
Ahhh, got it now! But still doesn't work..

Is this the correct setup:

image

 I'm getting profiler sample errors:

image

**Comment by nothke** [comment]
Correction: it works, but still getting these profiler errors, that's obviously a Unity issue as they come from the pipeline.

Btw, I'm on 2020.1 beta, using latest packages

**Comment by EMBYRDEV** [comment]
Yeah those are caused by Unity by the looks of it. I never touched the profiler and I wasnt getting any warnings on 2019.3

**Comment by harishl** [comment]
This is very useful. Thanks @MissEmbyr !
@goncalo Would be useful to have the alpha issues resolved. Any advice?

File: Resources/Shaders/DearImGui-Mesh.shader [hunk 1]
```diff
@@ -22,6 +22,28 @@
         }
     }
 
+	// shader for HD render pipeline
+	SubShader
+	{
+		Tags { "RenderType" = "Transparent" "RenderPipeline" = "HDRenderPipeline" "PreviewType" = "Plane" }
```
Comment (trapazza) [inline Resources/Shaders/DearImGui-Mesh.shader]: Just a totally uneducated guess: have you tried changing "Transparent" to "Opaque"?

**Comment by trapazza** [comment]
Hi @MissEmbyr  - Thanks for your work, really appreciated 🥇 

Have you figured out how to fix the apha issue? 
Also, in order you use your changes, should I just overwrite the contents of the original package with yours?

Thanks 👍

**Pull request closed without merging**